woocommerce/woocommerce-android
# [Product AI v2] generation prompt + option selector

Closes: #11972 Closes: #11973 


Please don't merge until the FluxC hash is updated.

### Description
This PR adds the following changes:
1. Logic for the new prompt that generates a product with three variants for the name/description.
2. UI and logic for the variant selector.
3. Refactorings to simplify the code of building the product properties rows.

⚠️ ~~The generation fails sometimes, this is because we still need to implement this part https://github.com/woocommerce/woocommerce-android/issues/11929, if you find issues with testing the PR, please let me know and I can suggest a temporary solution.~~
This was updated, the PR includes the changes of the PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3049 which implements the required change

### Testing information
1. Open the app and sign in using a Jeptack website.
2. Start product creation using AI.
3. Type something on the product features field.
4. Tap "Generate Product Details"
5. Confirm a product is generated according to the given details.
6. Notice the selector for the product variants.
7. Use the selector to change the product variant, and confirm it works as expected.

### Images/gif
**Note:** the buttons look bigger than what we have in Figma, that's because Compose enforces 48dp as the minimum size for touch targets, I believe we can override the minimum, but I don't think we should do it, the accessibility is more important than having it match Figma perfectly, WDYT?

<img src="https://github.com/user-attachments/assets/3d748b8e-73f9-4775-9300-c492c0ba2464" width=360 />
